src/main/java/net/sigmarik/abilitymod/AbilityMod.java

```diff
@@ -33,10 +40,99 @@ public class AbilityMod implements DedicatedServerModInitializer {
 
 	public static final double BOAT_ATTRACTION_FACTOR = 0.1;
 
+	public static final int FIRE_RESISTANCE_STATUS_EFFECT = 12;
+
+	public static final int FIRING_CHANCE_PER_TICK = 55;
+
 	public static final int ADDICTION_START_TIMER 	= 1800 * 20;	// 30 minutes
 	public static final int ADDICTION_MID_TIMER 	= 180 * 20;   	// 3 minutes
 	public static final int ADDICTION_WARNING_TIMER = 20 * 20;
 
+	public static final HashSet<Item> IRON_ITEMS = new HashSet<>(List.of(
+			Items.ANVIL,
+			Items.AXOLOTL_BUCKET,
+			Items.BLAST_FURNACE,
+			Items.BUCKET,
+			Items.CAULDRON,
+			Items.CHAIN,
+			Items.CHAINMAIL_BOOTS,
+			Items.CHAINMAIL_CHESTPLATE,
+			Items.CHAINMAIL_HELMET,
+			Items.CHAINMAIL_LEGGINGS,
+			Items.CHEST_MINECART,
+			Items.CHIPPED_ANVIL,
+			Items.COMPASS,
+			Items.CROSSBOW,
+			Items.DAMAGED_ANVIL,
+			Items.DEEPSLATE_IRON_ORE,
+			Items.DETECTOR_RAIL,
+			Items.FLINT_AND_STEEL,
+			Items.FURNACE_MINECART,
+			Items.HEAVY_WEIGHTED_PRESSURE_PLATE,
+			Items.HOPPER,
+			Items.HOPPER_MINECART,
+			Items.IRON_AXE,
+			Items.IRON_BARS,
+			Items.IRON_BLOCK,
+			Items.IRON_BOOTS,
+			Items.IRON_CHESTPLATE,
+			Items.IRON_DOOR,
+			Items.IRON_HELMET,
+			Items.IRON_HOE,
+			Items.IRON_HORSE_ARMOR,
+			Items.IRON_INGOT,
+			Items.IRON_LEGGINGS,
+			Items.IRON_NUGGET,
+			Items.IRON_ORE,
+			Items.IRON_PICKAXE,
+			Items.IRON_SHOVEL,
+			Items.IRON_SWORD,
+			Items.IRON_TRAPDOOR,
+			Items.LANTERN,
+			Items.LAVA_BUCKET,
+			Items.MILK_BUCKET,
+			Items.MINECART,
+			Items.PISTON,
+			Items.POWDER_SNOW_BUCKET,
+			Items.RAIL,
+			Items.RAW_IRON,
+			Items.RAW_IRON_BLOCK,
+			Items.SHEARS,
+			Items.SHIELD,
+			Items.SMITHING_TABLE,
+			Items.SOUL_LANTERN,
+			Items.STICKY_PISTON,
+			Items.STONECUTTER,
+			Items.TNT_MINECART,
+			Items.TRAPPED_CHEST,
+			Items.TRIPWIRE_HOOK,
+			Items.WATER_BUCKET
+	));
+
+	public static final HashSet<EntityType> NEUTRAL_MOBS = new HashSet<>(List.of(
+			EntityType.BEE,
+			EntityType.CAVE_SPIDER,
+			EntityType.DOLPHIN,
+			EntityType.ENDERMAN,
+			EntityType.GOAT,
+			EntityType.IRON_GOLEM,
+			EntityType.LLAMA,
+			EntityType.PANDA,
+			EntityType.PIGLIN,
+			EntityType.POLAR_BEAR,
+			EntityType.SPIDER,
+			EntityType.WOLF,
+			EntityType.ZOMBIFIED_PIGLIN,
+			EntityType.TRADER_LLAMA
+	));
+
+	public static final double FAST_COEFFICIENT = 0.3;
+	public static final UUID FAST_MODIFIER_UUID = new UUID(Random.createLocal().nextLong(), Random.createLocal().nextLong());
```

Comment:
Using random numbers as unique identifiers is a practice you use when you want your game to crash in random ways at random moments.

Please see if there is a way to generate truly unique UUID without affecting other important game systems, such as player account identification.